lib/phoenix/channel.ex

```diff
@@ -124,17 +124,34 @@ defmodule Phoenix.Channel do
                                                                      {:leave, Socket.t} |
                                                                      {:error, reason :: term, Socket.t}
 
-  defmacro __using__(_options) do
+  defmacro __using__(options \\ []) do
     quote do
       @behaviour unquote(__MODULE__)
-      import unquote(__MODULE__)
+      @pubsub_server unquote(options[:pubsub_server])
+      import unquote(__MODULE__), except: [broadcast: 4,
```

Comment:
I would use `:only` because there is less chance of accidentally importing something we don't want.